duecredit/io.py

```diff
@@ -154,6 +154,29 @@ def _model_citations(self, tags=None):
                 target_dict = packages
             target_dict['citations'][path].append(citation)
             target_dict['entry_keys'][path].append(entry_key)
+
+        # now we need to filter out the packages that don't have modules
+        # or objects cited
+        cited_packages = list(packages['citations'].keys())
+        cited_modules = list(modules['citations'].keys())
```

Comment:
no need for explicit `.keys()` here.  `list` would just try to traverse the container, so would already go through the dictionary keys:

```python
>>> list({1:2, 'a':[1,2]})
['a', 1]
```